### PR TITLE
az-snp/tdx-vtpm-verifier: add PCRs to claims map

### PR DESCRIPTION
**Pull request by mkulke**
PCR values are added in a `"tpm": { "pcr0": ..., "pcrN": ... }` hierarchy to the claims map so they can be compared to reference values.

### CODE REVIEW

File: attestation-service/verifier/src/az_snp_vtpm/mod.rs [hunk 1]
```diff
@@ -43,6 +44,24 @@ impl AzSnpVtpm {
...
+pub(crate) fn extend_claim_with_tpm_quote(
+    claim: &mut TeeEvidenceParsedClaim,
+    quote: &Quote,
+) -> Result<()> {
+    let Value::Object(ref mut map) = claim else {
+        bail!("failed to extend the claim, not an object");
+    };
+
+    let mut tpm_values = serde_json::Map::new();
+    for (i, pcr) in quote.pcrs_sha256().enumerate() {
```
Comment (portersrc) [inline attestation-service/verifier/src/az_snp_vtpm/mod.rs:56]: where's `pcrs_sha256` coming from? I couldn't track it down.

File: attestation-service/verifier/src/az_snp_vtpm/mod.rs [hunk 2]
```diff
@@ -43,6 +44,24 @@ impl AzSnpVtpm {
...
+        tpm_values.insert(format!("pcr{:02}", i), Value::String(hex::encode(pcr)));
+    }
+    debug!("extending claim with TPM quote: {:#?}", tpm_values);
+    map.insert("tpm".to_string(), Value::Object(tpm_values));
```
Comment (Xynnn007) [inline attestation-service/verifier/src/az_snp_vtpm/mod.rs:60]: We'd better add a document about the parsed claims for az-x-tpm platforms like [SGX](https://github.com/confidential-containers/kbs/blob/main/attestation-service/docs/parsed_claims.md#intel-sgx).

**Review by Xynnn007 — APPROVED** [review]
LGTM. Nice PR!

File: attestation-service/verifier/src/az_snp_vtpm/mod.rs [hunk 3]
```diff
@@ -43,6 +44,24 @@ impl AzSnpVtpm {
...
+    claim: &mut TeeEvidenceParsedClaim,
+    quote: &Quote,
+) -> Result<()> {
+    let Value::Object(ref mut map) = claim else {
+        bail!("failed to extend the claim, not an object");
+    };
+
+    let mut tpm_values = serde_json::Map::new();
+    for (i, pcr) in quote.pcrs_sha256().enumerate() {
+        tpm_values.insert(format!("pcr{:02}", i), Value::String(hex::encode(pcr)));
```
Comment (Xynnn007) [inline attestation-service/verifier/src/az_snp_vtpm/mod.rs:57]: this placeholder will set pcrs like
```json
{
    "pcr01": "xxx",
    ...
    "pcr13": "xxx"
}
```

with prefix-ed `0`s. Is this intended?

Comment (mkulke) [reply attestation-service/verifier/src/az_snp_vtpm/mod.rs:57]: yes, that's intended (.e.g for key sorting)

File: attestation-service/verifier/src/az_snp_vtpm/mod.rs (hunk 2 above)
Comment (mkulke) [reply attestation-service/verifier/src/az_snp_vtpm/mod.rs:60]: makes sense, will add documentation.

**Review by fitzthum — APPROVED** [review]
LGTM. I've merged https://github.com/confidential-containers/guest-components/pull/486

Do we need to bump the version in the client Cargo file to pick that up?

File: attestation-service/verifier/src/az_snp_vtpm/mod.rs (hunk 1 above)
Comment (portersrc) [reply attestation-service/verifier/src/az_snp_vtpm/mod.rs:56]: Thanks @mkulke, that info is helpful. Yeah I was sanity-checking the ordering when I asked this question, I believe. LGTM, nice PR.

**Pull request merged**